common/src/main/java/com/microsoft/identity/common/internal/cache/ActiveBrokerCacheUpdater.kt

```diff
@@ -76,20 +85,42 @@ class ActiveBrokerCacheUpdater(
             bundle.putString(ACTIVE_BROKER_SIGNING_CERTIFICATE_THUMBPRINT_KEY,
                 activeBroker.signingCertificateThumbprint)
         }
+
+        /**
+         * Adds the "broker discovery is now disabled" to the result bundle.
+         *
+         * @param bundle       The result bundle.
+         */
+        @JvmStatic
+        fun appendBrokerDiscoveryDisabledToResultBundle(bundle: Bundle) {
+            bundle.putBoolean(BROKER_DISCOVERY_DISABLED_KEY, true)
+        }
     }
 
     /**
-     * If the active broker is returned with the result bundle,
-     * update [IActiveBrokerCache] with it.
+     * If the active broker is returned with the result bundle, update [IActiveBrokerCache] with it.
+     * If the active broker indicates that the broker discovery is disabled, wipe [IActiveBrokerCache].
      *
-     * @param bundle       The result bundle. could be null (for backward compatibility).
+     * @param bundle    The result bundle. could be null (for backward compatibility).
      */
     fun updateCachedActiveBrokerFromResultBundle(bundle: Bundle?) {
         val methodTag = "$TAG:updateCachedActiveBrokerFromResultBundle"
         if (bundle == null) {
             return
         }
 
+        val shouldWipeCachedActiveBroker = bundle.getBoolean(BROKER_DISCOVERY_DISABLED_KEY, false)
+        if (shouldWipeCachedActiveBroker) {
+            Logger.info(methodTag, "Got a response indicating that the broker discovery is disabled." +
+                    "Will also wipe the local active broker cache," +
+                    "and skip broker discovery via IPC (only fall back to AccountManager) for the next 60 minutes.")
```

Comment:
We expect it to work again after 60 minutes?

Reply:
The feature might be turned on again in the next 60 mins (meaning, try the broker discovery again when that time comes. if it's still disabled, then wait for another 60 mins before the next try)